### DISCUSSION TITLE
couchdb not starting 'inet_tcp': register/listen error: epmd_close

### DISCUSSION THREAD

**Discussion by riker65** [apache/couchdb#3492] [Q&A]
Hi 

strangwise I now have an issue with couchdb.

`sudo -i -u couchdb /home/couchdb/bin/couchdb`

'inet_tcp': register/listen error: epmd_close

`epmd -debug
epmd: Tue Apr  6 11:31:09 2021: epmd running - daemon = 0
epmd: Tue Apr  6 11:31:09 2021: there is already a epmd running at port 4369
`

`epmd -names
epmd: up and running on port 4369 with data:`

killing epmd does not help

anyone having an idea?

**Comment by florianb** [comment]
HI @riker65,

> `there is already a epmd running at port 4369`

it seems like there's already an Erlang instance running on that port. This might be a foreign Erlang or Elixir application or a former CouchDb process. I'd suggest to check your task or process monitor if there is anything Erlang-based running in the background.

I hope this helps.

**Comment by riker65** [comment]
Hi Florian, 

I checked, but I only have an epmd running on tcp ipv6 I think I need another one for tcp ipv4?
wondering why.

I killed epmd with kill and restarted couchdb, but same error message.

` sudo netstat -nltp | grep -E '(4369|5672|25672)'

tcp6       0      0 :::4369                 :::*                    LISTEN      1/init
`

that is why I assume tcp 0.0.0.0 :4369 should be listening.

but just my assumption.

Thanks

**Reply by florianb** [reply]
You're welcome - do you know which process is claiming port 4369 (`netstat -tulpn`)?

The Erlang Port Mapper Daemon listens per default on all available interfaces. It should be possible (haven't tried that yet) to define a specific address (via `-address` in the `etc/vm.args`) to listen on.

But that wouldn't explain why you have it listening on IPv6 and not IPv4 as well.

**Reply by florianb** [reply]
And just for your interest, you're also able to configure the EPMD via environment variables: http://erlang.org/doc/man/epmd.html#environment-variables

**Reply by riker65** [reply]
Hi thanks again.

couchdb worked before I installed RaspAP 

I assume dnsmasq is blocking / missing something  to get couchdb started

`netstat -tulpn
Active Internet connections (only servers)

Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name

tcp        0      0 0.0.0.0:5900            0.0.0.0:*               LISTEN      474/vncserver-x11-c
tcp        0      0 0.0.0.0:81              0.0.0.0:*               LISTEN      595/lighttpd
tcp        0      0 127.0.0.1:53            0.0.0.0:*               LISTEN      1863/dnsmasq
tcp        0      0 10.3.141.1:53           0.0.0.0:*               LISTEN      1863/dnsmasq
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      534/sshd
tcp        0      0 127.0.0.1:631           0.0.0.0:*               LISTEN      1305/cupsd
tcp        0      0 0.0.0.0:1880            0.0.0.0:*               LISTEN      384/node-red
tcp        0      0 127.0.0.1:8088          0.0.0.0:*               LISTEN      468/influxd
tcp6       0      0 :::5900                 :::*                    LISTEN      474/vncserver-x11-c
tcp6       0      0 :::8080                 :::*                    LISTEN      466/java
tcp6       0      0 :::81                   :::*                    LISTEN      595/lighttpd
tcp6       0      0 :::4369                 :::*                    LISTEN      1/init
tcp6       0      0 fe80::59e8:bfb9:8d61:53 :::*                    LISTEN      1863/dnsmasq
tcp6       0      0 ::1:53                  :::*                    LISTEN      1863/dnsmasq
tcp6       0      0 :::8086                 :::*                    LISTEN      468/influxd
tcp6       0      0 :::22                   :::*                    LISTEN      534/sshd
tcp6       0      0 ::1:631                 :::*                    LISTEN      1305/cupsd
tcp6       0      0 :::3000                 :::*                    LISTEN      464/grafana-server
udp        0      0 0.0.0.0:35999           0.0.0.0:*                           327/avahi-daemon: r
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           327/avahi-daemon: r
udp        0      0 127.0.0.1:53            0.0.0.0:*                           1863/dnsmasq
udp        0      0 10.3.141.1:53           0.0.0.0:*                           1863/dnsmasq
udp        0      0 0.0.0.0:67              0.0.0.0:*                           1863/dnsmasq
udp        0      0 0.0.0.0:68              0.0.0.0:*                           1724/dhcpcd
udp        0      0 0.0.0.0:631             0.0.0.0:*                           450/cups-browsed
udp6       0      0 :::5353                 :::*                                327/avahi-daemon: r
udp6       0      0 :::546                  :::*                                1724/dhcpcd
udp6       0      0 fe80::59e8:bfb9:8d61:53 :::*                                1863/dnsmasq
udp6       0      0 ::1:53                  :::*                                1863/dnsmasq
udp6       0      0 :::40512                :::*                                327/avahi-daemon: r
`


trying to start at localhost with address option gives:

`epmd -d -address 127.0.0.1

epmd: Wed Apr  7 06:14:21 2021: epmd running - daemon = 0
epmd: Wed Apr  7 06:14:21 2021: there is already a epmd running at port 4369
`

and 
`

 sudo -i -u couchdb /home/couchdb/bin/couchdb

Protocol 'inet_tcp': register/listen error: epmd_close
`

I am confused

**Reply by florianb** [reply]
Okay - it seems like systemd is listening on that port. You might try `systemctl -all list-sockets` and see if this exposes a given service listening on the port 4369.

It is an important information that you run RaspAP on the same host. I don't know RaspAP but it seems like it manages iptables and dns services to provide its functionality. I would guess this adds a lot of complexity to the network stack and might interfere you system configuration in unexpected ways.

**Reply by riker65** [reply]
Hi Florian,

epmd start with option -systemd as a service, that is probably why it looks like systemd. 


`epmd -names
epmd: up and running on port 4369 with data:
`

Have to check what RaspAP did to the network stack which causes the error to couchdb.

Reinstalling couchdb does not  solve the issue.

Still checking 

Thanks a lot so far

**Reply by florianb** [reply]
You are welcome - i hope you're able to track that issue down.

**Comment by riker65** [comment]
No still not working
Will follow up
Was busy with other topics
